src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/services/AuditService.kt

```diff
@@ -5,25 +5,27 @@ import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.hmppsauditapi.config.trackEvent
+import uk.gov.justice.digital.hmpps.hmppsauditapi.jpa.AuditRepository
 import uk.gov.justice.digital.hmpps.hmppsauditapi.listeners.HMPPSAuditListener.AuditEvent
 
 @Service
-class AuditService(private val telemetryClient: TelemetryClient) {
+class AuditService(private val telemetryClient: TelemetryClient, private val auditRepository: AuditRepository) {
   companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
   fun audit(auditEvent: AuditEvent) {
     log.info("About to audit $auditEvent")
     telemetryClient.trackEvent("hmpps-audit", auditEvent.asMap())
+    auditRepository.save(auditEvent)
   }
 }
 
 private fun AuditEvent.asMap(): Map<String, String> {
-  var items = mapOf("what" to this.what, "when" to this.`when`)
+  var items = mapOf("what" to this.what, "when" to this.`when`.toString())
+  this.operationId?.let { items = items + ("operationId" to it) }
```

Comment:
`let` implies some transformation has occurred so I wonder if this would read better with a `run`?

Reply:
how about 
`items = items + this.operationId?.let {("operationId" to it)} ?: mapOf()  `